app/src/main/java/uk/gov/onelogin/MainActivityViewModel.kt

```diff
@@ -26,55 +26,52 @@ class MainActivityViewModel @Inject constructor(
 
     fun handleIntent(
         intent: Intent?,
-        context: Context
+        sharedPrefs: SharedPreferences
     ) {
         val data = intent?.data
         when {
-            data != null -> handleActivityResult(context, intent)
-            tokensAvailable(context) -> _next.value = HomeRoutes.START
+            data != null -> handleActivityResult(sharedPrefs, intent)
+            tokensAvailable(sharedPrefs) -> _next.value = HomeRoutes.START
```

Comment:
Do we need to check if `credChecker.isDeviceSecure()` when they resume to the app and before using the tokens again?

Reply:
It's a good point and possibly a question for product. Because this is part of the initial first time user setup check what level of authentication we can ask for.
Will raise with Dylan